mypy/fixup.py

```diff
@@ -247,6 +248,9 @@ def visit_type_var(self, tvt: TypeVarType) -> None:
         if tvt.upper_bound is not None:
             tvt.upper_bound.accept(self)
 
+    def visit_param_spec(self, p: ParamSpecType) -> None:
+        pass  # Nothing to descend into.
```

Comment:
What about `.upper_bound`? Also why it is ignored in similar fine-grained related logic?

Reply:
I hadn't updated some places after I added `upper_bound`. It's now processed here and fine-grained logic.